controllers/cluster_controller.go

```diff
@@ -384,21 +416,26 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, log logr.Logger
 	return ctrl.Result{}, nil
 }
 
-func (r *ClusterReconciler) ensureClusterOwnerReferences(ctx context.Context, clus *anywherev1.Cluster) error {
+func (r *ClusterReconciler) getChildObjectsForCluster(ctx context.Context, clus *anywherev1.Cluster) ([]kubernetes.Object, error) {
 	builder := cluster.NewDefaultConfigClientBuilder()
 	config, err := builder.Build(ctx, clientutil.NewKubeClient(r.client), clus)
 	if err != nil {
 		var notFound apierrors.APIStatus
 		if apierrors.IsNotFound(err) && errors.As(err, &notFound) {
 			clus.Status.FailureMessage = ptr.String(fmt.Sprintf("Dependent cluster objects don't exist: %s", notFound))
 		}
-		return err
+		return nil, err
 	}
 
 	childObjs := config.ChildObjects()
+
+	return childObjs, nil
+}
+
+func (r *ClusterReconciler) ensureClusterOwnerReferences(ctx context.Context, clus *anywherev1.Cluster, childObjs []kubernetes.Object) error {
```

Comment:
I would take the clusterConfig as input, which contains both the cluster and the child objs

Reply:
Done.